Can't mount bucket

I'm trying to mount the bucket locally using the following command:

gcsfuse -o nonempty --key-file /home/user/key.json --log-file /home/user/bucket/a.log --implicit-dirs warden-retail /home/user/bucket

but I'm getting this message

Using mount point: /home/user/bucket
Opening GCS connection...
daemonize.Run: readFromProcess: Decode: EOF
 
Any ideas?

I used the same command in a kubernetes pod and it works.

Running
<code> gcsfuse  --foreground  --key-file /home/user/key.json  bucket-name /home/user/bucket</code>

I get this:

<code>
Using mount point: /home/user/bucket
Opening GCS connection...
panic: Cannot connect to GCS: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.

goroutine 1 [running]:
github.com/googlecloudplatform/gcsfuse/internal/gcsx.NewBucketManager(0x0, 0x0, 0x0, 0x0, 0xbff0000000000000, 0x4014000000000000, 0x1000, 0xdf8475800, 0x200000, 0xc48c41, ...)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/internal/gcsx/bucket_manager.go:91 +0x237
main.mountWithConn(0xd29040, 0xc0000384c8, 0x7fffcb26a12b, 0xd, 0x7fffcb26a139, 0x14, 0xc00013e000, 0xd17e00, 0xc0001108a0, 0xc00012a320, ...)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/mount.go:94 +0x232
main.mountWithArgs(0x7fffcb26a12b, 0xd, 0x7fffcb26a139, 0x14, 0xc00013e000, 0xc00012a320, 0x1, 0x28, 0x0)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/main.go:280 +0x1e5
main.runCLIApp(0xc00013c000, 0x0, 0x0)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/main.go:408 +0xca5
main.run.func1(0xc00013c000)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/main.go:444 +0x34
github.com/googlecloudplatform/gcsfuse/vendor/github.com/urfave/cli.HandleAction(0xb126a0, 0xc0001220f0, 0xc00013c000, 0x0, 0x0)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/vendor/github.com/urfave/cli/app.go:528 +0x58
github.com/googlecloudplatform/gcsfuse/vendor/github.com/urfave/cli.(*App).Run(0xc000130000, 0xc0000321e0, 0x6, 0x6, 0x0, 0x0)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/vendor/github.com/urfave/cli/app.go:288 +0x655
main.run(0xc000000000, 0xc76d90)
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/main.go:448 +0xce
main.main()
	/tmp/build_gcsfuse_gopath715650293/src/github.com/googlecloudplatform/gcsfuse/main.go:466 +0x6f
</code>

I'm using the same key that's working on the kubernetes pod

--- a/internal/gcsx/bucket_manager.go
+++ b/internal/gcsx/bucket_manager.go
@@ -23,13 +23,11 @@ import (
 
 	"golang.org/x/net/context"
 
-	"cloud.google.com/go/storage"
 	"github.com/googlecloudplatform/gcsfuse/internal/canned"
 	"github.com/jacobsa/gcloud/gcs"
 	"github.com/jacobsa/gcloud/gcs/gcscaching"
 	"github.com/jacobsa/ratelimit"
 	"github.com/jacobsa/timeutil"
-	"google.golang.org/api/iterator"
 )
 
 type BucketConfig struct {
@@ -76,24 +74,17 @@ type BucketManager interface {
 
 type bucketManager struct {
 	config BucketConfig
-	conn   gcs.Conn
+	conn   *Connection
 
 	// Garbage collector
 	gcCtx                 context.Context
 	stopGarbageCollecting func()
-
-	client *storage.Client
 }
 
-func NewBucketManager(config BucketConfig, conn gcs.Conn) BucketManager {
-	client, err := storage.NewClient(context.Background())
-	if err != nil {
-		panic(fmt.Sprintf("Cannot connect to GCS: %v", err))
-	}
+func NewBucketManager(config BucketConfig, conn *Connection) BucketManager {
 	bm := &bucketManager{
 		config: config,
 		conn:   conn,
-		client: client,
 	}
 	bm.gcCtx, bm.stopGarbageCollecting = context.WithCancel(context.Background())
 	return bm
@@ -247,21 +238,6 @@ func (bm *bucketManager) ShutDown() {
 func (bm *bucketManager) ListBuckets(ctx context.Context) (
 	names []string,
 	err error) {
-	if bm.config.BillingProject == "" {
-		err = fmt.Errorf("ListBuckets: BillingProject not configured")
-		return
-	}
-	it := bm.client.Buckets(ctx, bm.config.BillingProject)
-	for {
-		bucket, nextErr := it.Next()
-		switch nextErr {
-		case nil:
-			names = append(names, bucket.Name)
-		case iterator.Done:
-			return
-		default:
-			err = nextErr
-			return
-		}
-	}
+	names, err = bm.conn.ListBuckets(ctx, bm.config.BillingProject)
+	return
 }

--- a/internal/gcsx/connection.go
+++ b/internal/gcsx/connection.go
@@ -1,0 +1,77 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcsx
+
+import (
+	"fmt"
+
+	"cloud.google.com/go/storage"
+	"github.com/jacobsa/gcloud/gcs"
+	"golang.org/x/net/context"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+type Connection struct {
+	wrapped gcs.Conn
+	gcs     *storage.Client
+}
+
+func NewConnection(cfg *gcs.ConnConfig) (c *Connection, err error) {
+	wrapped, err := gcs.NewConn(cfg)
+	if err != nil {
+		err = fmt.Errorf("Cannot create Conn: %v", err)
+		return
+	}
+
+	gcs, err := storage.NewClient(
+		context.Background(),
+		option.WithTokenSource(cfg.TokenSource))
+	if err != nil {
+		err = fmt.Errorf("Cannot create GCS client: %v", err)
+		return
+	}
+
+	c = &Connection{
+		wrapped: wrapped,
+		gcs:     gcs,
+	}
+	return
+}
+
+func (c *Connection) OpenBucket(
+	ctx context.Context,
+	options *gcs.OpenBucketOptions) (b gcs.Bucket, err error) {
+	b, err = c.wrapped.OpenBucket(ctx, options)
+	return
+}
+
+func (c *Connection) ListBuckets(
+	ctx context.Context,
+	projectId string) (names []string, err error) {
+	it := c.gcs.Buckets(ctx, projectId)
+	for {
+		bucket, nextErr := it.Next()
+		switch nextErr {
+		case nil:
+			names = append(names, bucket.Name)
+		case iterator.Done:
+			return
+		default:
+			err = nextErr
+			return
+		}
+	}
+}

--- a/main.go
+++ b/main.go
@@ -38,6 +38,7 @@ import (
 
 	"github.com/googlecloudplatform/gcsfuse/internal/auth"
 	"github.com/googlecloudplatform/gcsfuse/internal/canned"
+	"github.com/googlecloudplatform/gcsfuse/internal/gcsx"
 	"github.com/googlecloudplatform/gcsfuse/internal/logfile"
 	"github.com/jacobsa/daemonize"
 	"github.com/jacobsa/fuse"
@@ -168,7 +169,7 @@ func handleMemoryProfileSignals() {
 	}
 }
 
-func getConn(flags *flagStorage) (c gcs.Conn, err error) {
+func getConn(flags *flagStorage) (c *gcsx.Connection, err error) {
 	tokenSrc, err := auth.GetTokenSource(context.Background(), flags.KeyFile)
 	if err != nil {
 		err = fmt.Errorf("GetTokenSource: %v", err)
@@ -205,7 +206,7 @@ func getConn(flags *flagStorage) (c gcs.Conn, err error) {
 		cfg.GCSDebugLogger = log.New(os.Stdout, "gcs: ", log.Flags())
 	}
 
-	return gcs.NewConn(cfg)
+	return gcsx.NewConnection(cfg)
 }
 
 ////////////////////////////////////////////////////////////////////////
@@ -227,7 +228,7 @@ func mountWithArgs(
 	//
 	// Special case: if we're mounting the fake bucket, we don't need an actual
 	// connection.
-	var conn gcs.Conn
+	var conn *gcsx.Connection
 	if bucketName != canned.FakeBucketName {
 		mountStatus.Println("Opening GCS connection...")
 

--- a/mount.go
+++ b/mount.go
@@ -26,7 +26,6 @@ import (
 	"github.com/googlecloudplatform/gcsfuse/internal/perms"
 	"github.com/jacobsa/fuse"
 	"github.com/jacobsa/fuse/fsutil"
-	"github.com/jacobsa/gcloud/gcs"
 	"github.com/jacobsa/timeutil"
 )
 
@@ -37,7 +36,7 @@ func mountWithConn(
 	bucketName string,
 	mountPoint string,
 	flags *flagStorage,
-	conn gcs.Conn,
+	conn *gcsx.Connection,
 	status *log.Logger) (mfs *fuse.MountedFileSystem, err error) {
 	// Sanity check: make sure the temporary directory exists and is writable
 	// currently. This gives a better user experience than harder to debug EIO

